hack/validate/yamllint

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${SCRIPTDIR}/.validate"
+
+if [ -n "${TEST_FORCE_VALIDATE:-}" ]; then
+	files=(docs/api/*.yaml)
+else
+	IFS=$'\n'
+	files=($(validate_diff --diff-filter=ACMR --name-only -- docs/*.yaml || true))
+	unset IFS
+fi
+
+# validate the yamllint configuration file before anything else
+if out=$(yamllint -f parsable -d "{extends: default, rules: {document-start: disable}}" hack/validate/yamllint.yaml); then
```

Comment:
Should this be `$SCRIPTDIR` like the one right below this is supposed to be protecting?

Reply:
Had to think about that a few times, because the one below uses relative path for the file we're checking, but I guess you mean; if we don't use `$SCRIPTDIR` for the location then (theoretically) we could be validating a different file than is used below, right?

So, hm.. perhaps we should yes; let me update